### PR TITLE
Use `.clang-format` and reformat `scanner.c`

### PR DESCRIPTION
**Pull request by DavisVaughan**

### CODE REVIEW

File: src/scanner.c [hunk 1]
```diff
@@ -449,30 +456,31 @@ static bool scan(TSLexer* lexer, Stack* stack, const bool* valid_symbols) {
...
   bool debug = is_debug_build();
   return stack_new(debug);
 }
 
-bool tree_sitter_r_external_scanner_scan(void *payload,
-                                         TSLexer *lexer,
-                                         const bool *valid_symbols) {
+bool tree_sitter_r_external_scanner_scan(
+    void* payload, TSLexer* lexer, const bool* valid_symbols
+) {
```
Comment (kevinushey) [inline src/scanner.c:470]: Can we configure clang-format to prefer vertical arguments for this sort of code?

File: src/scanner.c [hunk 2]
```diff
@@ -449,30 +456,31 @@ static bool scan(TSLexer* lexer, Stack* stack, const bool* valid_symbols) {
...
 
-unsigned tree_sitter_r_external_scanner_serialize(void *payload, char *buffer) {
+unsigned tree_sitter_r_external_scanner_serialize(void* payload, char* buffer) {
   return stack_serialize(payload, buffer);
 }
 
-void tree_sitter_r_external_scanner_deserialize(void *payload, const char *buffer, unsigned length) {
+void tree_sitter_r_external_scanner_deserialize(
+    void* payload, const char* buffer, unsigned length
+) {
```
Comment (kevinushey) [inline src/scanner.c:480]: Same here

**Review by kevinushey — APPROVED** [review]
I'm overall on board but I hope we can keep function arguments either:

- On the same line as the function definition (if short enough), or
- Each on its own line.

**Comment by kevinushey** [comment]
Potentially related:

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#alignafteropenbracket

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#allowallargumentsonnextline

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#binpackarguments

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#binpackparameters

File: src/scanner.c (hunk 1 above)
Comment (DavisVaughan) [reply src/scanner.c:470]: Screenshot 2024-03-07 at 12 23 47 PM

I can make it looks like this with a combination of:

```
BinPackParameters: false
AllowAllParametersOfDeclarationOnNextLine: false
```

How's that?

Comment (kevinushey) [reply src/scanner.c:470]: Thanks; this looks good to me!

**Pull request merged**